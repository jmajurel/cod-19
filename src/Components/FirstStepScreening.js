import React from "react";
import "./FirstStepScreeming.css";

class FirstStepScreening extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      symptoms: []
    };
  }

  handleSubmit = event => {
    event.preventDefault();
    console.log("form submited: ", this.state);
    this.props.onSubmit(this.state.symptoms);
  };

  handleChange = event => {
    const newSymptom = event.target.value;
    const isActive = event.target.checked;
    let symptoms = [...this.state.symptoms];
    const foundItemIdx = symptoms.findIndex(symptom => symptom === newSymptom);
    if (isActive) {
      if (foundItemIdx < 0) symptoms.push(newSymptom);
    } else {
      if (foundItemIdx >= 0)
        symptoms = symptoms.filter(symptom => symptom !== newSymptom);
    }
    this.setState({
      symptoms
    });
  };

  render() {
    return (
      <div className="card firstStepScreening">
        <h1>First Step </h1>
        <form onSubmit={this.handleSubmit}>
          <h3>Choose your symptoms (if any):</h3>
          <ul className="symptoms">
            <li>
              <span role="img">ðŸ¤’</span> Fever
              <input
                className="symptom"
                type="checkbox"
                name="fever"
                value="fever"
                onChange={this.handleChange}
              />
            </li>
            <li>
              <span role="img">ðŸ˜«</span> Tiredness
              <input
                className="symptom"
                type="checkbox"
                name="tiredness"
                value="tiredness"
                onChange={this.handleChange}
              />
            </li>
            <li>
              <span role="img">ðŸ¤¯</span> Headache
              <input
                className="symptom"
                type="checkbox"
                name="headache"
                value="headache"
                onChange={this.handleChange}
              />
            </li>
            <li>
              <span role="img">ðŸ˜¤</span> Breathing issue
              <input
                className="symptom"
                type="checkbox"
                name="breathingIssue"
                value="breathingIssue"
                onChange={this.handleChange}
              />
            </li>
            <li>
              <span role="img">ðŸ¤§</span> Coughing/sore throat
              <input
                className="symptom"
                type="checkbox"
                name="coughingSoreThroat"
                value="coughingSoreThroat"
                onChange={this.handleChange}
              />
            </li>
          </ul>
          <button type="submit">Next</button>
        </form>
      </div>
    );
  }
}
export default FirstStepScreening;
