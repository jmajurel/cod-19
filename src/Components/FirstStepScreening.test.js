import React from "react";
import FirstStepScreening from "./FirstStepScreening";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { render } from "react-dom";

it("renders correctly", () => {
  const tree = renderer.create(<FirstStepScreening />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("displays symptoms", () => {
  const container = document.createElement("div");
  const mockSymptoms = [
    { _id: 1, name: "Fever" },
    { _id: 2, name: "Sore throat" },
    { _id: 3, name: "Runny nose" },
  ];

  act(() => {
    render(<FirstStepScreening symptoms={mockSymptoms} />, container);
  });
  const symptomList = container.querySelector("li");
  expect(mockSymptoms.map((symptom) => symptom.name)).toContain(
    symptomList.textContent.slice(3)
  );
});
