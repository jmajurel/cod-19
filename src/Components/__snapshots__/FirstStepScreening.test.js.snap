// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card firstStepScreening"
>
  <h1>
    firstStepScreening.title
     
  </h1>
  <form
    onSubmit={[Function]}
  >
    <h3>
      firstStepScreening.subTitle
    </h3>
    <ul
      className="symptoms"
    />
    <button
      type="submit"
    >
      firstStepScreening.nextBtn
    </button>
  </form>
</div>
`;
